common/genericResourceURLParts.go

```diff
@@ -44,6 +45,27 @@ func NewGenericResourceURLParts(resourceURL url.URL, location Location) GenericR
 	return g
 }
 
+func (g GenericResourceURLParts) GetAccountName() string {
```

Comment:
I'm uneasy about this function.  It introduces a new assumption about the structure of URLs into our code (i.e. that they will be formed in the way we expect).  That makes me nervous because for [blob custom domains](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-custom-domain-name?tabs=azure-portal) the assumption isn't actually true.  But perhaps more importantly, it in introduces a uniqueness-bug:

Imagine this command line, copying from a local copy of the Azurite emulator (which supports custom account names) to real storage

`azcopy copy http://127.0.0.1:10000/accountfoo https://accountfoo.blob.core.windows.net`

The source and the destination have the same account names... but they are not the same account.   

Admittedly I don't think Azurite currently supports Files endpoints.  But it still makes me nervous about this approach.

I'm not certain what the best approach is.  Ideas that come to mind are:
1. How about using something that, instead, gives all the URL parts up to (and including) the container/share name.   That will introduce no new assumptions about URL structure, and will guarantee uniquness.   OR
2. How about returning everything up to the end of the account name in IP style URLs, and for other ones, return the host name with "core.windows.net" removed (for compactness).  Function would need to be named something like GetUniqueAccountString or something....

Reply:
as per offline discussion, just handling the sddl caching via the URL sans blob path and SAS token is what was decided upon.